src/racket/dependency-graph.rkt

```diff
@@ -56,6 +56,16 @@
 (define (get-ids dg)
   (dep-graph-id-list dg))
 
+;; dg: dep-graph
+;; Returns if there is a path from from-id to to-id
+;; Note returns #f when from-id equals to to-id
+(define (path? dg to-id from-id)
+  #;(printf "finding path from ~a to ~a~%" from-id to-id)
+  (define-values (dist _) (bfs (dep-graph-graph dg) from-id))
+  (if (eq? to-id from-id)
+      #f
```

Comment:
Instead of (if c #f e), prefer (and c e) when the output is a boolean.

In other cases, if you are using #f to indicate that there is no solution, and e would produce a non-boolean value, then prefer the if form.

The codebase right now follows the first rule but not the second, but I want to start following the second rule too.